File: week01/example/ltc26_remove_duplicates_from_sorted_array.cpp
//
// Created by chenbingwei on 2022/4/14.
//
#include <vector>
#include <cassert>

int removeDuplicates(std::vector<int> &nums) {
    int n = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (i == 0 || nums[i] > nums[i - 1]) {
            nums[n] = nums[i];
            n++;
        }
    }
    return n;
}


int main() {
    std::vector<int> nums = {1, 2, 2, 3};
    std::vector<int> expectedNums = {1, 2, 3};

    int k = removeDuplicates(nums);
    assert(k == expectedNums.size());

    for (int i = 0; i < k; i++) {
        assert(nums[i] == expectedNums[i]);
    }
    return 0;
}

File: week01/example/ltc26_remove_duplicates_from_sorted_array.py
def removeDuplicates(nums):
    n = 0
    for i in range(len(nums)):
        if i == 0 or nums[i] > nums[i - 1]:
            nums[n] = nums[i]
            n += 1
    return n


if __name__ == '__main__':
    nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
    expect_nums = [0, 1, 2, 3, 4]

    k = removeDuplicates(nums)
    assert k == len(expect_nums)
    for i in range(k):
        assert nums[i] == expect_nums[i]
